### PR TITLE
Feature: Add LapseRate extensions

### PR DESCRIPTION
**Pull request by jnyrup**
This PR extends LapseRate with four operator overloads:

* TemperatureDelta  / LapseRate -> Length
* Temperature  / LapseRate -> Length
* Length * LapseRate -> TemperatureDelta
* LapseRate * Length -> TemperatureDelta

### CODE REVIEW

**Comment by ferittuncer** [comment]
Welcome and thanks for the contribution. I couldn't find any errors but let's wait for @angularsen for merging.

**Comment by jnyrup** [comment]
What are your gut feelings about the `Temperature / LapseRate -> Length`?
Will it cause more harm than good based on previous confusion related to the difference between `Temperature` and `TemperatureDelta`.

And similar, how about conversions to LapseRate?

I think `TemperatureDelta / Length -> LapseRate` makes perfect sense, while `Temperature / Length -> LapseRate` might  too ambiguous?

For consistency I could imagine we wanted to either only use `TemperatureDelta` or use `Temperature` in both conversions from and to `LapseRate`.

Any preferences on whether `TemperatureDelta / Length -> LapseRate` is an extension to `TemperatureDelta` or `Length`?

**Comment by angularsen** [comment]
If I understand correctly, I think `Temperature` cannot be used for the same reason you can't add two temperatures, you can only add a delta (range on scale) to a temperature (point on scale) or add two deltas to get a new larger delta.

`LapseRate [▲°C/m]` describes the gradient for how much the temperature changes (delta) over a distance, so the proposed equation becomes:

```
TemperatureDelta [▲°C] / LapseRate [▲°C/m] = 
TemperatureDelta [▲°C] / (TemperatureDelta [▲°C] / Length [m]) = 
Length [m]
```

In hindsight, LapseRate abbreviation should probably be changed to reflect this? It is currently missing the delta symbol. But not sure if this goes against the convention in the domain.

> Any preferences on whether TemperatureDelta / Length -> LapseRate is an extension to TemperatureDelta or Length?

I would say TemperatureDelta, because it is the leading quantity.

File: UnitsNet/CustomCode/Quantities/LapseRate.extra.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#endif
+    {
+        // Windows Runtime Component does not allow operator overloads: https://msdn.microsoft.com/en-us/library/br230301.aspx
+#if !WINDOWS_UWP
+        public static Length operator /(TemperatureDelta left, LapseRate right)
+        {
+            return Length.FromKilometers(left.KelvinsDelta / right.DegreesCelciusPerKilometer);
+        }
+
+        public static Length operator /(Temperature left, LapseRate right)
```
Comment (angularsen) [inline UnitsNet/CustomCode/Quantities/LapseRate.extra.cs]: I vote to drop this.

Comment (angularsen) [reply UnitsNet/CustomCode/Quantities/LapseRate.extra.cs]: If this is removed, I think this PR is good to go.

**Comment by jnyrup** [comment]
Thanks for the review. 
I've updated the PR with two changes: 

* removed the `Temperate / LapseRate -> Length` 
* added `TemperatureDelta / Length -> LapseRate`

Regarding the abbreviation:

* Currently it is `°C/m` for `LapseRateUnit.DegreeCelsiusPerKilometer`, which is wrong due to a missing a `k`.
* `▲°C/km` would be fine with me.

**Comment by ferittuncer** [comment]
@jnyrup About missing k, yeah I did implement that and when I changed base unit forgot to change the abbreviation also, my bad. 

And yeah, we should change the abbreviation to `▲°C/km`

**Comment by jnyrup** [comment]
And related, why do we use ▲ instead of ∆?

**Comment by ferittuncer** [comment]
I'm not aware of a specific reason for using ▲. I used ▲ because @angularsen used it in his comment above. Let's wait for his answer.

**Comment by angularsen** [comment]
No preference for me other than it already being used for TemperatureDelta, I suppose ∆ is more correct.

**Comment by ferittuncer** [comment]
Hmm, then I'm opening an issue for replacing every occurance of ▲ with ∆ in the library, for the sake of consistency.

I also see some minor differences in code-style in various places (talking in general, not about this p.r.). Perhaps we can set a standard on this and write it to wiki. Standards make contributing less confusing and reviewing less difficult, I believe.

**Comment by angularsen** [comment]
That would be good! And maybe adding it as a contribution guideline so it pops up whenever someone creates a new issue / PR.

**Comment by jnyrup** [comment]
I've updated the PR so the unit of lapse rate is now `▲°C/km`.

**Comment by angularsen** [comment]
@jnyrup Shouldn't we use the new delta symbol? Per #328 .

**Comment by jnyrup** [comment]
I thought that was the job of #327?

**Comment by angularsen** [comment]
Sure, but if you are already modifying the abbreviation here, why not use the correct symbol to begin with?

**Comment by jnyrup** [comment]
The PR has been updated with the correct delta symbol.
Thanks for the fast reviews!

**Comment by angularsen** [comment]
Thanks! Nuget 3.81 is out.

**Pull request merged**